build.gradle.kts

```diff
@@ -18,9 +18,108 @@ java {
+tasks.register<Exec>("installBuf") {
+    description = "Installs the Buf CLI."
+    environment("GOBIN", bufCLIFile.parentFile.absolutePath)
+    outputs.file(bufCLIFile)
+    commandLine("go", "install", "github.com/bufbuild/buf/cmd/buf@latest")
+}
+
+tasks.register<Exec>("installLicenseHeader") {
+    description = "Installs the Buf license-header CLI."
+    environment("GOBIN", bufLicenseHeaderCLIFile.parentFile.absolutePath)
+    outputs.file(bufLicenseHeaderCLIFile)
+    commandLine("go", "install", "github.com/bufbuild/buf/private/pkg/licenseheader/cmd/license-header@latest")
+}
+
+tasks.register<Exec>("licenseHeader") {
+    dependsOn("installLicenseHeader")
+    description = "Runs the Buf license-header CLI."
+    commandLine(
+        bufLicenseHeaderCLIPath,
+        "--license-type",
+        "apache",
+        "--copyright-holder",
+        "Buf Technologies, Inc.",
+        "--year-range",
+        project.findProperty("license-header.years")!!.toString(),
+        "--ignore",
+        "src/main/java/build/buf/validate/",
+        "--ignore",
+        "conformance/src/main/java/build/buf/validate/conformance/",
+    )
+}
+
+tasks.register<Exec>("generateTestSourcesImports") {
+    dependsOn("installBuf")
+    description = "Generates code with buf generate --include-imports for unit tests."
+    commandLine(
+        bufCLIPath,
+        "generate",
+        "--template",
+        "src/test/resources/proto/buf.gen.imports.yaml",
+        "src/test/resources/proto",
+        "--include-imports",
+    )
+}
+
+tasks.register<Exec>("generateTestSourcesNoImports") {
+    dependsOn("installBuf")
+    description = "Generates code with buf generate --include-imports for unit tests."
+    commandLine(bufCLIPath, "generate", "--template", "src/test/resources/proto/buf.gen.noimports.yaml", "src/test/resources/proto")
+}
+
+tasks.register("generateTestSources") {
+    dependsOn("generateTestSourcesImports", "generateTestSourcesNoImports")
+    description = "Generates code with buf generate for unit tests"
+}
+
+tasks.register<Exec>("exportProtovalidateModule") {
+    dependsOn("installBuf")
+    description = "Exports the bufbuild/protovalidate module sources to src/main/resources."
+    commandLine(
+        bufCLIPath,
+        "export",
+        "buf.build/bufbuild/protovalidate:${project.findProperty("protovalidate.version")}",
+        "--output",
+        "src/main/resources",
+    )
+}
+
+tasks.register<Exec>("generateSources") {
+    dependsOn("installBuf")
+    description = "Generates sources for the bufbuild/protovalidate module sources to src/main/java."
+    commandLine(bufCLIPath, "generate", "--template", "buf.gen.yaml", "src/main/resources")
+}
+
+tasks.register<Exec>("generateConformance") {
+    dependsOn("installBuf")
+    description = "Generates sources for the bufbuild/protovalidate-testing module to conformance/src/main/java."
+    commandLine(bufCLIPath, "generate", "--template", "buf.gen.yaml", "src/main/resources")
```

Comment:
I might be missing it, but this looks like the same `buf generate` call as on line 97 - I would think these would be different?

in the makefile they are:

```
	$(BIN)/buf generate --template buf.gen.yaml src/main/resources
	$(BIN)/buf generate --template conformance/buf.gen.yaml -o conformance/ buf.build/bufbuild/protovalidate-testing:$(PROTOVALIDATE_VERSION)
```

Reply:
Good eye - fixed. Verified by removing the generated code and re-running `./gradlew generate` that it re-added it.